### PR TITLE
[CSL-509] Make sure it's possible to supply multiple filters with autocomplete requests

### PR DESCRIPTION
**Pull request by esezen**

### CODE REVIEW

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIOAutocompleteTest.java [hunk 1]
```diff
@@ -82,4 +84,27 @@ public void autocompleteShouldReturnAResultWithHiddenFields() throws Exception {
...
+
+    @Test
+    public void autocompleteShouldReturnAResultWithOneFilter() throws Exception {
+        ConstructorIO constructor = new ConstructorIO("", apiKey, true, null);
+        UserInfo userInfo = new UserInfo(3, "c62a-2a09-faie");
+        AutocompleteRequest request = new AutocompleteRequest("item1");
+        request.getFilters().put("group_id", Arrays.asList("All"));
+        AutocompleteResponse response = constructor.autocomplete(request, userInfo);
+        assertEquals("autocomplete product suggestions exist", response.getSections().get("Products").size(), 2);
+        assertTrue("autocomplete result id exists", response.getResultId() != null);
```
Comment (sblaurock) [inline constructorio-client/src/test/java/io/constructor/client/ConstructorIOAutocompleteTest.java:96]: Would it be possible to check if the filters were set correctly in the response?

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIOAutocompleteTest.java [hunk 2]
```diff
@@ -82,4 +84,27 @@ public void autocompleteShouldReturnAResultWithHiddenFields() throws Exception {
...
+    @Test
+    public void autocompleteShouldReturnAResultWithMultipleFilters() throws Exception {
+        ConstructorIO constructor = new ConstructorIO("", apiKey, true, null);
+        UserInfo userInfo = new UserInfo(3, "c62a-2a09-faie");
+        AutocompleteRequest request = new AutocompleteRequest("item1");
+        request.getFilters().put("group_id", Arrays.asList("All"));
+        request.getFilters().put("Brand", Arrays.asList("XYZ"));
+        AutocompleteResponse response = constructor.autocomplete(request, userInfo);
+        assertEquals("autocomplete product suggestions exist", response.getSections().get("Products").size(), 1);
+        assertTrue("autocomplete result id exists", response.getResultId() != null);
```
Comment (sblaurock) [inline constructorio-client/src/test/java/io/constructor/client/ConstructorIOAutocompleteTest.java:108]: Same as above.

**Review by sblaurock — COMMENTED** [review]
@esezen was it not possible to specify filters at all previously? Nice work!

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIOAutocompleteTest.java (hunk 1 above)
Comment (esezen) [reply constructorio-client/src/test/java/io/constructor/client/ConstructorIOAutocompleteTest.java:99]: I just added those but I am not sure if it's the right way to do it. I couldn't find any examples in the current tests. I will tag Zubin to make sure they are good

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIOAutocompleteTest.java [hunk 3]
```diff
@@ -94,6 +97,7 @@ public void autocompleteShouldReturnAResultWithOneFilter() throws Exception {
         AutocompleteResponse response = constructor.autocomplete(request, userInfo);
         assertEquals("autocomplete product suggestions exist", response.getSections().get("Products").size(), 2);
         assertTrue("autocomplete result id exists", response.getResultId() != null);
+        assertEquals("autocomplete request [group_id] filter should match", ((ArrayList)((LinkedTreeMap)response.getRequest().get("filters")).get("group_id")).get(0), "All");
```
Comment (esezen) [inline constructorio-client/src/test/java/io/constructor/client/ConstructorIOAutocompleteTest.java:100]: @rubikzube Does this look good?

**Review by rubikzube — APPROVED** [review]
This looks good for the moment.  In the future we probably want to update `response.getRequest()` to be more strongly typed than it is at the moment, which should make the tests a bit easier to understand.  Good casting tho.

**Pull request merged**